tools/lou_checkyaml.c

```diff
@@ -282,6 +282,82 @@ read_mode(yaml_parser_t *parser) {
 }
 
 int *
+read_inPos(yaml_parser_t *parser, int len) {
+	int *pos = malloc(sizeof(int) * len);
+	int i = 0;
+	yaml_event_t event;
+	int parse_error = 1;
+	char *tail;
+
+	if (!yaml_parser_parse(parser, &event) || (event.type != YAML_SEQUENCE_START_EVENT))
+		yaml_error(YAML_SEQUENCE_START_EVENT, &event);
+	yaml_event_delete(&event);
+
+	while ((parse_error = yaml_parser_parse(parser, &event)) &&
+			(event.type == YAML_SCALAR_EVENT)) {
+		errno = 0;
+		int val = strtol((const char *)event.data.scalar.value, &tail, 0);
+		if (errno != 0)
+			error_at_line(EXIT_FAILURE, 0, file_name, event.start_mark.line + 1,
+					"Not a valid inpos position '%s'. Must be a number\n",
+					event.data.scalar.value);
+		if ((const char *)event.data.scalar.value == tail)
+			error_at_line(EXIT_FAILURE, 0, file_name, event.start_mark.line + 1,
+					"No digits found in inpos position '%s'. Must be a number\n",
+					event.data.scalar.value);
+		pos[i++] = val;
```

Comment:
This is a buffer overrun waiting to happen. Never read more than you allocated. Add a condition to the while loop (`i` has to be smaller than `len`).

Reply:
It was code that I took from the original read_cursorPos() function. I will have a look at it.

Reply:
Ah, hehe. I guess at the time I hadn't been burned by the CVEs. Yes looking at it it looks like it has the same problems. Basically it should be smth along the line of 
``` C
  while ((parse_error = yaml_parser_parse(parser, &event)) &&
	 (event.type == YAML_SCALAR_EVENT) &&
	 (i < len)) {
```
in all places